### PR TITLE
Gracefully handle SIGTERM and SIGINT

### PR DESCRIPTION
**Pull request by harry-cpp**
This needs a bit testing.

### CODE REVIEW

File: MonoGame.Framework/Platform/Utilities/SignalWatcher.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace MonoGame.Utilities
+{
+    public static class SignalWatcher
```
Comment (tomspilman) [inline MonoGame.Framework/Platform/Utilities/SignalWatcher.cs]: Might want to make this private/internal?

Comment (cra0zy) [reply MonoGame.Framework/Platform/Utilities/SignalWatcher.cs]: Ups, yes.

**Comment by Jjagg** [comment]
@harry-cpp What's the status of this?

**Comment by harry-cpp** [comment]
Works on Linux, haven't tested it on Windows / Mac.

**Pull request closed without merging**